lib/modem_info/modem_info.c

```diff
@@ -299,7 +385,11 @@ int modem_info_short_get(enum modem_info info, u16_t *buf)
 		return -EIO;
 	}
 
-	err = modem_info_parse(modem_data[info], recv_buf);
+	if (info != MODEM_INFO_FW_VERSION) {
+		cmd_length = modem_info_remove_cmd(recv_buf);
```

Comment:
`modem_info_remove_cmd` can return negative values, yet `cmd_length` is `size_t` - this should be sorted out. Here and below.

Reply:
Fixed.